src/redux/wallet/actions.js

```diff
@@ -29,10 +30,16 @@ export const watchTransfer = (notice: TransferNoticeModel, token: AbstractTokenD
   dispatch({type: WALLET_TRANSACTION, tx: notice.tx()})
 }
 
-export const watchInitWallet = () => async (dispatch) => {
+// TODO @ipavlenko: Refactor this, provide ability to detach watchers
+export const watchInitWallet = () => async (dispatch, getState) => {
+
+  const state = getState()
+  const profile = state.get('session').profile
+
   dispatch({type: WALLET_TOKENS_FETCH})
   const dao = await contractsManagerDAO.getERC20ManagerDAO()
-  let tokens = await dao.getUserTokens() // TODO @bshevchenko: put here user tokens addresses
+  // TODO @ipavlenko: ETH Token have no address
+  let tokens = await dao.getUserTokens(profile.tokens().toArray().filter(address => address != null))
```

Comment:
profile should not contain ETH and TIME tokens addresses, because this two tokens are obligatory for everyone